src/main/webapp/query-control.js

```diff
@@ -119,4 +119,48 @@ function replaceValueIfEmpty(dataListId) {
   if (typedSoFar === '') {
     inputlist.value = storedVal;
   }
-}
\ No newline at end of file
+}
+
+// Sort location dropdown alphabetically
+function sortStateDropdownList() {
+  // Adapted from w3schools
+  const list = document.getElementById('location');
+  let switching = true;
+  let i;
+  while (switching) {
+    switching = false;
+    options = list.getElementsByTagName('option');
+    // all states option always first
+    for (i = 1; i < (options.length - 1); i++) {
+      shouldSwitch = false;
+      if (options[i].value.toLowerCase() > options[i + 1].value.toLowerCase()) {
+        shouldSwitch = true;
+        break;
+      }
+    }
+    if (shouldSwitch) {
+      options[i].parentNode.insertBefore(options[i + 1], options[i]);
+      switching = true;
+    }
+  }
+}
+
+function createStateDropdownList() {
+  const datalist = document.getElementById('location');
+  let optionElem = document.createElement('option');
+  optionElem.value = 'each U.S. state';
+  optionElem.setAttribute('data-value', 'state');
+  datalist.appendChild(optionElem);
+
+  for (const state in stateInfo) {
+    if (stateInfo.hasOwnProperty(state)) {
+      if (state !== '72') { // Skip Puerto Rico
```

Comment:
instead can you just remove this entry from stateinfo file.

Reply:
The Census API returns Puerto Rico as a state and I couldn't find a way to query all states excluding Puerto Rico. So, I wasn't able to remove PR from state-info without breaking the each U.S. state query.